src/python/dart/web/api/action.py

```diff
@@ -123,6 +148,25 @@ def patch_action(action):
     p = JsonPatch(request.get_json())
     return update_action(action, Action.from_dict(p.apply(action.to_dict())))
 
+def should_update(new_state, current_state):
+    ''' A new state of action should only be updated if it is a more 'advanced' state.
+        The order is PENDING < RUNNABLE < STARTING < RUNNING < COMPLETED < {FAILED|SUCCEEDED}.
+
+        E.g. if current_state is PENDING and new_state is RUNNING then we should update.
+             if current_state is SUCCEEDED and new_state is RUNNING then we should NOT update.
```

Comment:
I am sending the state of the Action (Job in batch terms) each time it changes a state.  This is done via ECS stream event => lambda => SQS queue => lambda that reads the parsed state of the job.  Since events might be sent out of order or be duplicated we should make sure we only update the state of the action "going forward", i.e. if an action is in FAILED mode it should not move back to RUNNING mode.